### ISSUE TITLE
ReadableBuffer is incompatible with AnyStr

### ISSUE THREAD

**Issue by ehandal** [microsoft/pyright#1633]
Code for repro:
```
with open('read', 'rb') as fr, open('write', 'wb') as fw:
    shutil.copyfileobj(fr, fw)
```
Error:
```
error: Argument of type "BufferedWriter" cannot be assigned to parameter "fdst" of type "SupportsWrite[AnyStr@copyfileobj]" in function "copyfileobj"
    TypeVar "_T_contra@SupportsWrite" is contravariant
      Type "ReadableBuffer" is incompatible with constrained type "AnyStr" (reportGeneralTypeIssues)
```
Running pyright 1.1.121 on the command line.

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This change is included in pyright 1.1.122, which I just published. It will also be in the next release of pylance.
